src/dosext/mfs/mfs.c

```diff
@@ -4200,11 +4189,7 @@ static int dos_fs_redirect(struct vm86_regs *state)
       if ((larg.l_start & mask) != 0)
         larg.l_start = (larg.l_start & ~mask) | ((larg.l_start & mask) >> 2);
 
-#ifdef F_SETLK64
-      ret = lock_file_region(fd, F_SETLK64, &larg, pt->offset, pt->size);
-      if (ret == -1 && errno == EINVAL)
-#endif
-        ret = lock_file_region(fd, F_SETLK, &larg, larg.l_start, larg.l_len);
+      ret = lock_file_region(fd, F_SETLK, &larg, pt->offset, pt->size);
```

Comment:
Is this okay, it seems the fallback to 32bit was using different start and len values?

Reply:
0x100000000LL was some magic value to indicate the lock
presence, and for 32bit it was truncated. Its not something
that was passed from DOS, so the actual value doesn't matter.
This is my understanding. I never really looked into that code.